main/src/ca/uwaterloo/flix/language/phase/typer/TypeReduction.scala

```diff
@@ -293,6 +293,37 @@ object TypeReduction {
       case (Type.Apply(Type.Apply(Type.Cst(TypeConstructor.Array, _), elmType1, _), rcVar1, _),
       Type.Apply(Type.Apply(Type.Cst(TypeConstructor.Array, _), elmType2, _), rcVar2, _)) =>
         isSubtype(elmType1, elmType2)
+      // Arrow to Java function interface
+      // TODO INTEROP: generics support
+      case (Type.Apply(Type.Apply(Type.Apply(Type.Cst(TypeConstructor.Arrow(2), _), eff, _), var_arg, _), var_ret, _), Type.Cst(TypeConstructor.Native(clazz), _)) =>
+        (var_arg, var_ret) match {
+          case (Type.Cst(tc1, _), Type.Cst(tc2, _)) =>
+            (tc1, tc2) match {
+              // IntStream
+              case (TypeConstructor.Int32, TypeConstructor.Unit) =>
+                clazz == classOf[java.util.function.IntConsumer]
+              case (TypeConstructor.Int32, TypeConstructor.Bool) =>
+                clazz == classOf[java.util.function.IntPredicate]
+              case (TypeConstructor.Int32, TypeConstructor.Int32) =>
+                clazz == classOf[java.util.function.IntUnaryOperator]
+              // DoubleStream
+              case (TypeConstructor.Float64, TypeConstructor.Unit) =>
+                clazz == classOf[java.util.function.DoubleConsumer]
+              case (TypeConstructor.Float64, TypeConstructor.Bool) =>
+                clazz == classOf[java.util.function.DoublePredicate]
+              case (TypeConstructor.Float64, TypeConstructor.Float64) =>
+                clazz == classOf[java.util.function.DoubleUnaryOperator]
+              // LongStream
+              case (TypeConstructor.Int64, TypeConstructor.Unit) =>
+                clazz == classOf[java.util.function.LongConsumer]
+              case (TypeConstructor.Int64, TypeConstructor.Bool) =>
+                clazz == classOf[java.util.function.LongPredicate]
+              case (TypeConstructor.Int64, TypeConstructor.Int64) =>
+                clazz == classOf[java.util.function.LongUnaryOperator]
```

Comment:
Are these all the relevant functions?

Reply:
This should be. As they were supported in Resolver.scala.

Reply:
(for non-generic things)